crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -432,20 +432,34 @@ impl EventHandler {
                     "Intercepted HTLC"
                 );
 
+                let fake_channel_payments = self.fake_channel_payments.lock().unwrap();
+
+                let target_node_id = match fake_channel_payments.get(&requested_next_hop_scid) {
+                    None => {
+                        tracing::warn!(fake_scid = requested_next_hop_scid, "Could not forward the intercepted HTLC because we didn't have a node registered with said fake scid");
+
+                        if let Err(err) = self.channel_manager.fail_intercepted_htlc(intercept_id) {
+                            tracing::error!("Could not fail intercepted htlc {err:?}")
+                        }
+
+                        return Ok(());
+                    }
+                    Some(target_node_id) => target_node_id,
+                };
```

Comment:
🔧 It's generally good practice to drop a `MutexGuard` as soon as you're done with it, so that some other thread can access the resource immediately.

```suggestion
                let target_node_id = {
                    let fake_channel_payments = self.fake_channel_payments.lock().unwrap();
                    match fake_channel_payments.get(&requested_next_hop_scid) {
                        None => {
                            tracing::warn!(fake_scid = requested_next_hop_scid, "Could not forward the intercepted HTLC because we didn't have a node registered with said fake scid");

                            if let Err(err) =
                                self.channel_manager.fail_intercepted_htlc(intercept_id)
                            {
                                tracing::error!("Could not fail intercepted htlc {err:?}")
                            }

                            return Ok(());
                        }
                        Some(target_node_id) => *target_node_id,
                    }
                };
```

Definitely not a problem with your code as you've just moved code around, but it's something to keep in mind in general.